pkg/controller/toolchainconfig/cache_test.go

```diff
@@ -0,0 +1,163 @@
+
+	t.Run("return config that is stored in client", func(t *testing.T) {
+		// given
+		config := newToolchainConfigWithReset(t, AutomaticApprovalCfg().MaxUsersNumber(123, PerMemberClusterCfg("member1", 321)))
+		cl := NewFakeClient(t, config)
+
+		// when
+		actual, err := GetConfig(cl, HostOperatorNs)
+
+		// then
+		require.NoError(t, err)
+		assert.False(t, actual.AutomaticApproval().IsEnabled())
+		assert.Equal(t, 123, actual.AutomaticApproval().MaxNumberOfUsersOverall())
+		assert.Equal(t, config.Spec.Host.AutomaticApproval.MaxNumberOfUsers.SpecificPerMemberCluster, actual.AutomaticApproval().MaxNumberOfUsersSpecificPerMemberCluster())
+		assert.Equal(t, 0, actual.AutomaticApproval().ResourceCapacityThresholdDefault())
+		assert.Equal(t, config.Spec.Host.AutomaticApproval.ResourceCapacityThreshold.SpecificPerMemberCluster, actual.AutomaticApproval().ResourceCapacityThresholdSpecificPerMemberCluster())
+		assert.Equal(t, 3, actual.Deactivation().DeactivatingNotificationInDays())
+
+		t.Run("returns the same when the cache hasn't been updated", func(t *testing.T) {
+			// given
+			newConfig := newToolchainConfigWithReset(t, AutomaticApprovalCfg().MaxUsersNumber(666))
+			cl := NewFakeClient(t, newConfig)
+
+			// when
+			actual, err := GetConfig(cl, HostOperatorNs)
+
+			// then
+			require.NoError(t, err)
+			assert.False(t, actual.AutomaticApproval().IsEnabled())
+			assert.Equal(t, 123, actual.AutomaticApproval().MaxNumberOfUsersOverall())
+			assert.Equal(t, config.Spec.Host.AutomaticApproval.MaxNumberOfUsers.SpecificPerMemberCluster, actual.AutomaticApproval().MaxNumberOfUsersSpecificPerMemberCluster())
+			assert.Equal(t, 0, actual.AutomaticApproval().ResourceCapacityThresholdDefault())
+			assert.Equal(t, config.Spec.Host.AutomaticApproval.ResourceCapacityThreshold.SpecificPerMemberCluster, actual.AutomaticApproval().ResourceCapacityThresholdSpecificPerMemberCluster())
+			assert.Equal(t, 3, actual.Deactivation().DeactivatingNotificationInDays())
+		})
+
+		t.Run("returns the new config when the cache was updated", func(t *testing.T) {
+			// given
+			newConfig := newToolchainConfigWithReset(t, AutomaticApprovalCfg().MaxUsersNumber(666), DeactivationCfg().DeactivatingNotificationDays(5))
+			cl := NewFakeClient(t)
+
+			// when
+			updateConfig(newConfig)
+
+			// then
+			actual, err := GetConfig(cl, HostOperatorNs)
+			require.NoError(t, err)
+			assert.False(t, actual.AutomaticApproval().IsEnabled())
+			assert.Equal(t, 666, actual.AutomaticApproval().MaxNumberOfUsersOverall())
+			assert.Empty(t, actual.AutomaticApproval().MaxNumberOfUsersSpecificPerMemberCluster())
+			assert.Equal(t, 0, actual.AutomaticApproval().ResourceCapacityThresholdDefault())
+			assert.Empty(t, actual.AutomaticApproval().ResourceCapacityThresholdSpecificPerMemberCluster())
+			assert.Equal(t, 5, actual.Deactivation().DeactivatingNotificationInDays())
+		})
+	})
+}
+
+func TestGetConfigErrored(t *testing.T) {
```

Comment:
Do we have a test for the case where there is no ToolchainConfig resource with the expected name?

Reply:
Added it in https://github.com/codeready-toolchain/host-operator/pull/438/commits/c9cc996c362634d70059fe8b625c8fb3846b6623